packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/softphone.service.js

```diff
@@ -0,0 +1,23 @@
+import { SOFTPHONE_TYPE } from './softphone.constants';
+
+export default class SofpthoneService {
+  /* @ngInject */
+  constructor($http) {
+    this.$http = $http;
+  }
+
+  generateLink(billingAccount, serviceName, name) {
+    return this.$http
+      .post(`/telephony/${billingAccount}/line/${serviceName}/devices`, {
+        name,
+        type: SOFTPHONE_TYPE,
+      })
+      .then(({ data }) => data.deviceId)
+      .then((deviceId) =>
+        this.$http.post(
+          `/telephony/${billingAccount}/line/${serviceName}/devices/${deviceId}/enroll`,
+        ),
+      )
+      .then(({ data }) => data);
+  }
```

Comment:
I don't think it's a good idea to merge the 2 because the enroll will be necessary for the concept of re-registration. I think it's better to have the device creation call and enroll separately.

Users can generate the link several times if he wish without recreating a device